pkg/events/dep.go

```diff
@@ -0,0 +1,78 @@
+package events
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"github.com/aquasecurity/tracee/pkg/ebpf/probes"
+)
+
+//
+// Dependencies (events, ksyms, tailcalls & capabilities)
+//
+
+//
+// NOTE:
+//
+// Within the Dependencies type, there is a single field which thread-safety is guaranteed by the
+// field type itself, and not by Dependencies methods: Capabilities. This happens because there are
+// currently two options to chose when adding fields to Dependencies:
+//
+// 1. To have a complex type (array, map, slice, ...) as a Dependencies field and protect it with
+//    mutexes that are also internal to Dependencies.
+//
+// 2. To have a single atomic pointer dereference to a type instance pointer, and rely on its
+//    internal synchronization mechanisms (like the Dependencies type).
+//
+
+type Dependencies struct {
+	events        map[ID]struct{}                 // map[eventID]struct{}
+	probes        map[probes.Handle]*Probe        // map[handle]*ProbeDependency
```

Comment:
About being a pointer: the "probe dependency" has a "required/not_required" bool (with set/unset/read methods) that needs to be atomic (and singular) among all execution threads.

About the need for an instance: is "required/not_required" something we would change runtime? I believe it would when adding/removing policies). 

About the naming, unfortunately there is a similar problem to all "dep_XXX" types:

Event: is actually EventDefinition and not "an event" (confuses with Tracee Event). Probe is ProbeDependency. KSymbol is a KSymbolDependency. Capabilities are CapabilitiesNeeded.

So, at least for now, I kept:

1. event_group
2. event
3. type_XXX (for types used in dependencies)
4. dep_YYY (for methods of types used in dependencies)

But I'm opened to suggestions (so we create a pattern for all of them together, not a single one).